src/main.cpp

```diff
@@ -6062,23 +6068,41 @@ bool ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vRecv, in
     }
 
 
-    else if (strCommand == NetMsgType::XPEDITEDREQUEST)
+    else if (strCommand == NetMsgType::XPEDITEDREQUEST && IsThinBlocksEnabled())
     {
+        if (!pfrom->ThinBlockCapable())
+        {
+            LOCK(cs_main);
+            Misbehaving(pfrom->GetId(), 100);
+            return error("XPEDITEDREQUEST message received from a non thinblock node, peer=%d", pfrom->GetId());
+        }
```

Comment:
I continue to believe we're banning without good cause.  This is easily detected and not harmful, a low misbehaving seems more appropriate in this and several other cases.  Can we come up with a policy about what deserves a ban and apply it consistently?